Faq: WIP (#782)
oscar-system/Oscar.jl: docs/src/General/styleguide.md

- `CamelCase` for types, `snake_case` for everything else
- `isfoo` not `is_foo`
- for generic concepts choose generic names, based on general algebraic
  concepts, preferably not special names from your area of speciality.
- use Julia conventions where applicable.
- in Julia we have multiple dispatch, so we do not need a function like

>   ``point_from_matrix`` as the from part is clear by the type of the argument.
>   It should be called ``points(::Matrix)`` in some variation.
>   Similar: ``matrix_to_points``. It is fine to use them internally, where

Reviewer: ```suggestion
  `point_from_matrix` as the from part is clear by the type of the argument.
  It should be called `points(T::Matrix)` in some variation.
  Similar: `matrix_to_points`. It is fine to use them internally, where
```

  useful, however, internally.
- follow the mathematics. If your function needs a list of points, you should
  create a point-type and then use this. For user-facing functions, please do not
  use re-purposed lists, arrays, matrices...
- if already existing types in Oscar are almost what you need, consider

>   improving them over writing you own. The `Problem` is that, while it is

Reviewer: ```suggestion
  improving them instead of writing you own. While it might be
```
